### PR TITLE
JDBC hometask

### PR DESCRIPTION
**Pull request by AHGitHubAccount**
JDBC and Hibernate hometask

### CODE REVIEW

File: DatabaseConnection.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package com.example.demo.repository;
+
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Controller;
+
+import java.sql.*;
```
Comment (vova-pereb) [inline DatabaseConnection.java]: Please, do not make wildcard class import.

File: DatabaseConnection.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    @Value("${spring.datasource.username}")
+    private  String user;
+    @Value("${spring.datasource.driver-class-name}")
+    private String driver;
+    @Value("${spring.datasource.password}")
+    private String password;
+
+    // JDBC variables for opening and managing connection
+    private static Connection con;
+    private static Statement stmt;
```
Comment (vova-pereb) [inline DatabaseConnection.java]: 1) Unused variables.
2) Do not declare such class fields (Statement, ResultSet). These variables should be created on demand instead.

File: Email.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
+package com.example.demo;
+
+import java.io.Serializable;
+import java.util.Date;
+
+
+public class Email implements Serializable {
+
+    private String recipientName=null;
```
Comment (vova-pereb) [inline Email.java]: Read [Java Code Conventions](https://www.oracle.com/technetwork/java/codeconventions-150003.pdf)

File: EmailRepository.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.example.demo.Email;
+
+import java.sql.Date;
+import java.sql.SQLException;
+import java.util.ArrayList;
+
+public interface EmailRepository {
+
+    public ArrayList<Email> getActive() ;
```
Comment (vova-pereb) [inline EmailRepository.java]: All methods in interface are public by default.

File: JDBCEmailRepository.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+import java.sql.*;
+import java.util.ArrayList;
+
+@Repository
+public class JDBCEmailRepository implements EmailRepository {
+
+    private Connection connection;
+    private Statement statement=null;
+    private ResultSet resultSet=null;
+    private ArrayList<Email> emails=new ArrayList<Email>();
```
Comment (vova-pereb) [inline JDBCEmailRepository.java]: It is redundant to specify generic type on the right side, if generic is specified on the left side.

File: JDBCEmailRepository.java [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    }
+
+
+    public ArrayList<Email> getActive()  {
+        emails.clear();
+
+        try {
+
+            statement = connection.createStatement();
+            String selectActiveQuery="Select * from email where isSent=false";
```
Comment (vova-pereb) [inline JDBCEmailRepository.java]: Keep uppercase SQL code style.
It makes possible to mentally distinguish reserved words (commands) from identifiers.

```
SELECT * FROM email WHERE isSent = false
```

Comment (vova-pereb) [inline JDBCEmailRepository.java]: As I said, SELECT * statement is forbidden.

Comment (vova-pereb) [inline JDBCEmailRepository.java]: I recommend to check boolean values in such way: 
```
isSent IS FALSE
```

File: JDBCEmailRepository.java [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+      }catch (SQLException e){
+          e.printStackTrace();
+      }
+  }
+
+    @Override
+    public void updateDeliveryDateForOneEmail(int id, Date date) {
+
+        try{
+            String updateDeliveryDateForOneEmail="update Email set deliveryDate=(?) where idEmail=(?) ";
```
Comment (vova-pereb) [inline JDBCEmailRepository.java]: I recommend to use database fields name in snake_case.

File: repository/EmailRepository.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+@Repository
+public interface EmailRepository extends JpaRepository<Email,Long> {
+
+    ArrayList<Email> getAllByIsSentIsFalse() ;
+    ArrayList<Email> findAll();
+    void deleteById(Long id);
+    Email saveAndFlush(Email email);
+    @Query("UPDATE Email SET delivery_Date=date WHERE id=idForUpdate")
+    Email updateDeliveryDateForEmailByID(@Param("idForUpdate")Long id,@Param("date")Date date);
+    @Query("UPDATE Email SET isSent=TRUE WHERE id ")
```
Comment (AHGitHubAccount) [inline repository/EmailRepository.java]: Line 21: WHERE id=idForUpdate

**Pull request closed without merging**